build.gradle

```diff
@@ -39,8 +39,13 @@ dependencies {
     // api
     implementation 'io.javalin:javalin-bundle:3.13.6'
 
-    testImplementation 'org.junit.jupiter:junit-jupiter-api:5.7.0'
-    testRuntimeOnly 'org.junit.jupiter:junit-jupiter-engine:5.7.0'
+    // testing
+    testImplementation 'org.junit.jupiter:junit-jupiter-api:5.7.2'
+    testRuntimeOnly 'org.junit.jupiter:junit-jupiter-engine:5.7.2'
+    testImplementation 'org.assertj:assertj-core:3.19.0'
+    testImplementation 'org.testcontainers:testcontainers:1.15.1'
+    testImplementation 'org.testcontainers:junit-jupiter:1.15.1'
+    testImplementation 'org.testcontainers:postgresql:1.15.1'
```

Comment:
testcontainers is a fantastic set of libraries that spins up docker containers as part of your test suite. We're using it spin up a postgresql container so that we can run true database integration tests against it.

Reply:
Rad! I don't think I've heard of that for other languages.

Reply:
Here's their website. https://www.testcontainers.org/